src/sys/socket/sockopt.rs

```diff
@@ -42,25 +42,33 @@ const TCP_CA_NAME_MAX: usize = 16;
 /// * Type of the value that you are going to set.
 /// * Type that implements the `Set` trait for the type from the previous item (like `SetBool` for
 ///    `bool`, `SetUsize` for `usize`, etc.).
+#[macro_export]
 macro_rules! setsockopt_impl {
     ($name:ident, $level:expr, $flag:path, $ty:ty, $setter:ty) => {
         #[allow(deprecated)] // to allow we have deprecated socket option
-        impl SetSockOpt for $name {
+        impl $crate::sys::socket::SetSockOpt for $name {
             type Val = $ty;
 
-            fn set<F: AsFd>(&self, fd: &F, val: &$ty) -> Result<()> {
-                unsafe {
-                    let setter: $setter = Set::new(val);
-
-                    let res = libc::setsockopt(
+            fn set<F: std::os::unix::io::AsFd>(
+                &self,
+                fd: &F,
+                val: &$ty,
+            ) -> $crate::Result<()> {
+                use $crate::sys::socket::sockopt::Set;
+                let setter: $setter =
+                    $crate::sys::socket::sockopt::Set::new(val);
```

Comment:
Considering that it has already been imported:

```suggestion
                use $crate::sys::socket::sockopt::Set;
                let setter: $setter = Set::new(val);
```

Reply:
done